File: 프로그래머스/2/84512. 모음 사전/모음 사전.py
def dfs(input_str):
    global str_list 
    alpabet_list = ['A','E','I','O','U']
    str_list.append(input_str)
    for alpabet in alpabet_list : 
        new_str = input_str + alpabet
        if(len(new_str)>5):
            return
        else:
            dfs(new_str)
            
def solution(word):
    global str_list 
    str_list = []
    answer = 0
    dfs('')
    answer = str_list.index(word)
    return answer

File: 프로그래머스/2/84512. 모음 사전/모음 사전.js
function solution(word) {
    var answer = 0;
    const history = []
    dfs('',history);
    answer = history.indexOf(word) +1;
    return answer;
}

function dfs(path, history){
    if (path.length >5) return
    if (path) history.push(path)
    alphabet = ['A','E','I','O','U']
    for (let a of alphabet){
        dfs(path + a, history);
    }
}